## Travis tests have failed

Hey @ronish31,
Please read the following log in order to understand the failure reason.
It'll be awesome if you fix what's wrong and commit the changes.


### Node.js: lts&#x2F;*

<a href="https:&#x2F;&#x2F;travis-ci.org&#x2F;logzio&#x2F;logzio-nodejs&#x2F;jobs&#x2F;672013359">View build log</a>


<details>
  <summary>
    <strong>
     npm test
    </strong>
  </summary>

```
> logzio-nodejs@2.0.0 test /home/travis/build/logzio/logzio-nodejs
> jest

 FAIL  test/logger.test.js
  ● Console

    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Closing, purging messages.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Closing, purging messages.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #2
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #2 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Woke up and saw 3 messages to send. Sending now...
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #2
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #2 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - failed on error: RequestError: Error: ESOCKETTIMEDOUT
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - Trying again in 40[ms], attempt no. 2
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Bulk #1 - sent successfully
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending bulk #1

  ● logger › logs a single line › sends log as a string with an object

    assert.equal(received, expected)

    Expected value to be equal to:
      "message: {\"key1\":\"val1\",\"key2\":\"val2\"}"
    Received:
      "message: "

    Difference:

    - Expected
    + Received

    - message: {"key1":"val1","key2":"val2"}
    + message: 

      70 |             logger.log(strMsg, obj);
      71 | 
    > 72 |             assert.equal(logger._createBulk.getCall(0).args[0][0].message, expectedLog);
         |                    ^
      73 |             logger._createBulk.restore();
      74 |             logger.close();
      75 |         });

      at Object.<anonymous> (test/logger.test.js:72:20)

 PASS  test/udp.test.js
  ● Console

    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending messages via udp
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending messages via udp
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending messages via udp
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Closing, purging messages.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending messages via udp
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Buffer is full - sending bulk
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending messages via udp
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Starting to send messages via udp.
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Error while sending udp packets. err = udp error
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending last messages and closing...
    console.log lib/logzio-nodejs.js:239
      logzio-nodejs: Sending messages via udp


Test Suites: 1 failed, 1 passed, 2 total
Tests:       1 failed, 21 passed, 22 total
Snapshots:   0 total
Time:        2.099s
Ran all test suites.
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
npm ERR! Test failed.  See above for more details.
```

</details>


###### TravisBuddy Request Identifier: 510dc040-78b9-11ea-85c2-5bba50baba14

--- a/lib/logzio-nodejs.d.ts
+++ b/lib/logzio-nodejs.d.ts
@@ -20,7 +20,7 @@ interface ILoggerOptions {
 
 interface ILogzioLogger extends ILoggerOptions{
     jsonToString(json: string): string;
-    log(msg: string): void;
+    log(msg: string, obj?: object): void;
     close(): void;
     sendAndClose(callback: (error: Error, bulk: object) => void): void;
 }

--- a/lib/logzio-nodejs.js
+++ b/lib/logzio-nodejs.js
@@ -188,11 +188,11 @@ class LogzioLogger {
         }
     }
 
-    log(msg, obj = null) {
+    log(msg, obj) {
         if (this.closed === true) {
             throw new Error('Logging into a logger that has been closed!');
         }
-        if (obj){
+        if (![null, undefined].includes(obj)){
             msg += JSON.stringify(obj);
         }
         if (typeof msg === 'string') {

--- a/lib/logzio-nodejs.js
+++ b/lib/logzio-nodejs.js
@@ -188,9 +188,12 @@ class LogzioLogger {
         }
     }
 
-    log(msg) {
+    log(msg, obj = null) {
         if (this.closed === true) {
             throw new Error('Logging into a logger that has been closed!');
+        }
+        if (obj){
+            msg += JSON.stringify(obj);
         }
         if (typeof msg === 'string') {
             msg = {

--- a/test/logger.test.js
+++ b/test/logger.test.js
@@ -58,6 +58,22 @@ describe('logger', () => {
             logger.close();
         });
 
+        it('should send a log with an object as additional params', (done) => {
+            const logger = createLogger({
+                bufferSize: 1,
+                callback: done,
+            });
+            sinon.spy(logger, '_createBulk');
+            const obj = { key1: "val1", key2: "val2"};
+            const strMsg = 'message: ';
+            const expectedLog = strMsg + JSON.stringify(obj);
+            logger.log(strMsg, obj);
+
+            assert.equal(logger._createBulk.getCall(0).args[0][0].message, expectedLog);
+            logger._createBulk.restore();
+            logger.close();
+        });
+
         it('sends log as a string with extra fields', (done) => {
             const logger = createLogger({
                 bufferSize: 1,

--- a/test/logger.test.js
+++ b/test/logger.test.js
@@ -58,7 +58,7 @@ describe('logger', () => {
             logger.close();
         });
 
-        it('should send a log with an object as additional params', (done) => {
+        it('should send a log with an object as additional param', (done) => {
             const logger = createLogger({
                 bufferSize: 1,
                 callback: done,

